File: PlayWall/src/main/java/de/tobias/playpad/action/actions/cart/handler/PlayStopHandler.scala
package de.tobias.playpad.action.actions.cart.handler

import de.tobias.playpad.action.InputType
import de.tobias.playpad.action.actions.cart.CartAction
import de.tobias.playpad.pad.Pad
import de.tobias.playpad.project.Project

class PlayStopHandler extends CartActionHandler {

	override def performAction(`type`: InputType,
							   cartAction: CartAction,
							   pad: Pad,
							   project: Project): Unit = {
		if (`type` eq InputType.PRESSED) {
			if (pad.isPlay) {
				pad.stop()
			} else { // Allow the listener to send the feedback
				cartAction.getPadPositionListener.setSend(false)
				pad.play()
			}
		}
	}
}

File: PlayWall/src/main/java/de/tobias/playpad/action/actions/cart/handler/PlayStopHandler.java
package de.tobias.playpad.action.actions.cart.handler;

import de.thecodelabs.midi.event.KeyEventType;
import de.tobias.playpad.action.actions.CartAction;
import de.tobias.playpad.pad.Pad;

public class PlayStopHandler implements CartActionHandler
{
	@Override
	public void performAction(KeyEventType keyEventType, CartAction cartAction, Pad pad)
	{
		if(keyEventType.equals(KeyEventType.DOWN))
		{
			if(pad.isPlay())
			{
				pad.stop();
			}
			else
			{ // Allow the listener to send the feedback
				// cartAction.getPadPositionListener.setSend(false) TODO Fix
				pad.play();
			}
		}
	}
}
